src/languages/clickhouse/clickhouse.formatter.ts

```diff
@@ -0,0 +1,161 @@
+import { DialectOptions } from '../../dialect.js';
+import { expandPhrases } from '../../expandPhrases.js';
+import { functions } from './clickhouse.functions.js';
+import { dataTypes, keywords, keywordPhrases } from './clickhouse.keywords.js';
+
+const reservedSelect = expandPhrases(['SELECT [DISTINCT]']);
+
+const reservedClauses = expandPhrases([
+  // https://clickhouse.com/docs/sql-reference/statements/explain
+  'EXPLAIN [AST | SYNTAX | QUERY TREE | PLAN | PIPELINE | ESTIMATE | TABLE OVERRIDE]',
+]);
+
+const standardOnelineClauses = expandPhrases([
+  // https://clickhouse.com/docs/sql-reference/statements/create
+  'CREATE [OR REPLACE] [TEMPORARY] TABLE [IF NOT EXISTS]',
+  // https://clickhouse.com/docs/sql-reference/statements/update
+  'UPDATE',
+  // https://clickhouse.com/docs/sql-reference/statements/system
+  'SYSTEM RELOAD {DICTIONARIES | DICTIONARY | FUNCTIONS | FUNCTION | ASYNCHRONOUS METRICS} [ON CLUSTER]',
+  'SYSTEM DROP {DNS CACHE | MARK CACHE | ICEBERG METADATA CACHE | TEXT INDEX DICTIONARY CACHE | TEXT INDEX HEADER CACHE | TEXT INDEX POSTINGS CACHE | REPLICA | DATABASE REPLICA | UNCOMPRESSED CACHE | COMPILED EXPRESSION CACHE | QUERY CONDITION CACHE | QUERY CACHE | FORMAT SCHEMA CACHE | DROP FILESYSTEM CACHE}',
+  'SYSTEM FLUSH LOGS',
+  'SYSTEM RELOAD {CONFIG | USERS}',
+  'SYSTEM SHUTDOWN',
+  'SYSTEM KILL',
+  'SYSTEM FLUSH DISTRIBUTED',
+  'SYSTEM START DISTRIBUTED SENDS',
+  'SYSTEM {STOP | START} {LISTEN | MERGES | TTL MERGES | MOVES | FETCHES | REPLICATED SENDS | REPLICATION QUEUES | PULLING REPLICATION LOG}',
+  'SYSTEM {SYNC | RESTART | RESTORE} REPLICA',
+  'SYSTEM {SYNC | RESTORE} DATABASE REPLICA',
+  'SYSTEM RESTART REPLICAS',
+  'SYSTEM UNFREEZE',
+  'SYSTEM WAIT LOADING PARTS',
+  'SYSTEM {LOAD | UNLOAD} PRIMARY KEY',
+  'SYSTEM {STOP | START} [REPLICATED] VIEW',
+  'SYSTEM {STOP | START} VIEWS',
+  'SYSTEM {REFRESH | CANCEL | WAIT} VIEW',
+  // https://clickhouse.com/docs/sql-reference/statements/show
+  'SHOW [CREATE] {TABLE | TEMPORARY TABLE | DICTIONARY | VIEW | DATABASE}',
+  'SHOW DATABASES [[NOT] {LIKE | ILIKE}]',
+  'SHOW [FULL] [TEMPORARY] TABLES [FROM | IN]',
+  'SHOW [EXTENDED] [FULL] COLUMNS {FROM | IN}',
+  // https://clickhouse.com/docs/sql-reference/statements/attach
+  'ATTACH {TABLE | DICTIONARY | DATABASE} [IF NOT EXISTS]',
+  // https://clickhouse.com/docs/sql-reference/statements/detach
+  'DETACH {TABLE | DICTIONARY | DATABASE} [IF EXISTS]',
+  // https://clickhouse.com/docs/sql-reference/statements/drop
+  'DROP {DICTIONARY | DATABASE | USER | ROLE | QUOTA | PROFILE | SETTINGS PROFILE | VIEW | FUNCTION | NAMED COLLECTION | ROW POLICY | POLICY} [IF EXISTS]',
+  'DROP [TEMPORARY] TABLE [IF EXISTS] [IF EMPTY]',
+  // https://clickhouse.com/docs/sql-reference/statements/exists
+  'EXISTS [TEMPORARY] {TABLE | DICTIONARY | DATABASE}',
```

Comment:
The linked docs say that the `TABLE | DICTIONARY | DATABASE` part is optional.

But making all that optional would result in `EXISTS (...)` expressions being detected as `EXISTS` statements.

So I guess you made a deliberate choice here.